packages/fxa-settings/src/pages/Signup/container.tsx

```diff
@@ -75,22 +99,76 @@ const SignupContainer = ({
             hardNavigateToContentServer(
               `/signin?email=${queryParamModel.email}`
             );
+            // TODO: Probably move this to the Index page onsubmit once
+            // the index page is converted to React, we need to run it in
+            // signup and signin for Sync
+          } else if (isSyncWebChannel) {
+            firefox.fxaCanLinkAccount({ email: queryParamModel.email });
           }
+        } else if (isSyncWebChannel) {
+          // TODO: Probably move this to the Index page onsubmit once
+          // the index page is converted to React, we need to run it in
+          // signup and signin for Sync
+          firefox.fxaCanLinkAccount({ email: queryParamModel.email });
         }
       }
       setShowLoadingSpinner(false);
     })();
   });
 
+  useEffect(() => {
+    // This sends a webchannel message to the browser to prompt a response
+    // that we listen for.
+    // TODO: In content-server, we send this on app-start for all integration types.
```

Comment:
Maybe not in this PR while we work through some of the other issues, but I think it does make sense to add listeners when the integration itself is initalized.